lib/spree/core/product_duplicator.rb

```diff
@@ -24,7 +24,7 @@ def duplicate_product
           new_product.created_at = nil
           new_product.deleted_at = nil
           new_product.updated_at = nil
-          new_product.unit_value = true
+          new_product.unit_value = nil
```

Comment:
This is weird indeed. It was probably just done to fulfil the validation. The value is forwarded to the variant and there it's converted to `1.0`. So we could just set it to `1.0` here to be more clear.

Reply:
If the variant unit is weight or volume then we need to set this value. Otherwise the variant will be invalid and the cloning fails.